### PR TITLE
Cryo pods crutchy fix

### PR DESCRIPTION
**Pull request by BELNAKBU**
## About the PR

Just a little bit crutchy fix of cryo pods make them work somehow

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Cryo pods now works. somehow

### CODE REVIEW

File: Content.Server/Medical/CryoPodSystem.cs [hunk 1]
```diff
@@ -100,16 +102,25 @@ public override void Update(float frameTime)
...
+                if (targetDamage.TotalDamage > 0)
+                {
+                    var solutionToInject = _solutionContainerSystem.SplitSolution(container.Value, containerSolution, cryoPod.BeakerTransferAmount);
+                    _bloodstreamSystem.TryAddToChemicals(patient.Value, solutionToInject, bloodstream);
+                    _reactiveSystem.DoEntityReaction(patient.Value, solutionToInject, ReactionMethod.Injection);
+                }
+                else
+                {
+                    EjectBody(cryoPod.Owner, cryoPod);
+                    return;
```
Comment (vulppine) [inline Content.Server/Medical/CryoPodSystem.cs:122]: unneeded

File: Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs [hunk 2]
```diff
@@ -120,18 +121,49 @@ protected void AddAlternativeVerbs(EntityUid uid, SharedCryoPodComponent cryoPod
...
+            _blocker.CanMove(args.User))
+        {
+            AlternativeVerb verb = new();
+            verb.Act = () => InsertBody(uid, args.User, cryoPodComponent);
+            verb.Text = Loc.GetString("cryo-pod-verb-noun-occupant");
+            args.Verbs.Add(verb);
         }
     }
 
+    protected void AddInsertOtherVerb(EntityUid uid, SharedCryoPodComponent cryoPodComponent, GetVerbsEvent<InteractionVerb> args)
```
Comment (vulppine) [inline Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs:146]: while this technically works, this doesn't match the 'drag-drop' interaction pattern that SS14 (especially other medical entities) have - i'd say that this should be changed to match that, instead

File: Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs [hunk 3]
```diff
@@ -120,18 +121,49 @@ protected void AddAlternativeVerbs(EntityUid uid, SharedCryoPodComponent cryoPod
...
+        if (IsOccupied(cryoPodComponent))
         {
-            args.Verbs.Add(new AlternativeVerb
-            {
-                Text = Loc.GetString("cryo-pod-verb-noun-occupant"),
-                Category = VerbCategory.Eject,
-                Priority = 1, // Promote to top to make ejecting the ALT-click action
-                Act = () => TryEjectBody(uid, args.User, cryoPodComponent)
-            });
+            AlternativeVerb verb = new();
```
Comment (vulppine) [inline Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs:126]: do `var` as the type, and add the type in the constructor/`new()` statement instead

File: Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs [hunk 4]
```diff
@@ -154,7 +186,17 @@ protected void OnCryoPodPryInterrupted(EntityUid uid, SharedCryoPodComponent cry
     {
         cryoPodComponent.IsPrying = false;
     }
+    public bool IsOccupied(SharedCryoPodComponent scannerComponent)
```
Comment (vulppine) [inline Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs:189]: does this have to be public?

File: Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs [hunk 5]
```diff
@@ -154,7 +186,17 @@ protected void OnCryoPodPryInterrupted(EntityUid uid, SharedCryoPodComponent cry
...
+    {
+        return scannerComponent.BodyContainer.ContainedEntity != null;
+    }
+    public bool CanCryoPodInsert(EntityUid uid, EntityUid target, SharedCryoPodComponent? component = null)
```
Comment (vulppine) [inline Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs:193]: does this have to be public?

**Review by vulppine — CHANGES_REQUESTED** [review]
the changelog is too vague, please change it to something that actually references what you add/fix in this PR

File: Content.Server/Medical/CryoPodSystem.cs (hunk 1 above)
Comment (vulppine) [reply Content.Server/Medical/CryoPodSystem.cs:122]: (by this, i meant the return statement)

**Comment by metalgearsloth** [comment]
Re-open if you come back to it.

**Pull request closed without merging**